### DISCUSSION TITLE
Is it possible to create editable sequence diagram with ReactFlow?

### DISCUSSION THREAD

**Discussion by bogy0** [xyflow/xyflow#3121] [Q&A]
Hello everyone,

Our team is considering using the ReactFlow diagramming library for various diagrams. However, we require the capability to display a specific kind of diagram known as the Sequence diagram.

Sequence Diagrams show elements as they interact over time and are organized according to object (horizontally) and time (vertically).
The horizontal axis shows the elements or actors involved in the interaction. (for example, a "Browser" an "App Server" and an "Auth Server".

In the showcase section is an example by Stripe that resembles a sequence diagram. However, I would like some help constructing it based on the available documentation and example.
Here is the example by Stripe: https://stripe.com/docs/payments/checkout/how-checkout-works#lifecycle

In the Stripe example, we can see different diagram notations like `Actors`, `Lifelines`, `Activations`, `Call Message`, `Receive Message`, `Self Message`.

I'm looking for guidance on building sequence diagrams in ReactFlow. Would anyone from the ReactFlow team or from the community be able to provide some helpful hints?

I appreciate any help you can give.

Example screenshot about a sequence diagram:
![Screenshot 2023-06-06 at 16 50 02](https://github.com/wbkd/react-flow/assets/5119480/129adcb5-c635-46e3-bd58-399007b3488d)

**Comment by shobrook** [comment]
Bumping this. Lack of code examples for sequence diagrams is blocking my company from switching to React Flow from MermaidJS.

**Reply by moklick** [reply]
React Flow was not built for this in the first place but for editors. Currently we are working on the v12 release but something like a mermaid to ReactFlow transformer is something we would like to check out soon.

**Reply by shobrook** [reply]
I'm interested in implementing this myself. Is there an existing effort me and my engineering team could participate in?

**Comment by victorng8** [comment]
I was reading the https://stripe.com/blog/markdoc blog and it gave some sample code to generate the sequence diagram

```
{% diagram type="sequence" description="Usage-based billing" %}
  {% node #customer icon="customer" %}
  Customer
  {% /node %}
  {% node #typographic icon="platform" %}
  Typographic
  {% /node %}
  {% node #stripe icon="stripe" %}
  Stripe
  {% /node %}
  {% edge from="customer" to="typographic" %}
  Select plan
  {% /edge %}
  {% edge from="typographic" to="typographic" %}
  Create [usage records](/docs/api/usage_records)
  {% /edge %}
```

However I am not able to render the sample. Wish there was a quickstart guide on this..

**Reply by moklick** [reply]
They defined custom tags in order render the diagrams. Those custom tags are not published afaik.

**Comment by greendesertsnow** [comment]
I was using Mermaidjs for this, but it looks ugly.

**Comment by CSworenI** [comment]
It seems like this could be accomplished by utilizing [custom nodes](https://reactflow.dev/examples/nodes/custom-node), [custom edges](https://reactflow.dev/examples/edges/custom-edges), and [drag and drop](https://reactflow.dev/examples/interaction/drag-and-drop) in proper proportions. You might even be able to properly order it with one of the layout modifiers. They even provide annotation node examples in the [overview example](https://reactflow.dev/examples/overview).

I've been working pretty hard at this for the last month and am learning a lot about how to make the flows dynamic to our needs (high level workflow tracking with lower level workflows, nodes, and tasks associated with each node).

**Reply by rot169** [reply]
Hi, curious if you got anywhere with this? I was thinking of doing the same, but am starting from zero here so would love to reuse/extend any existing efforts!

**Comment by sebzuddas** [comment]
Did this implementation ever get published?
